### ISSUE TITLE
Partially valid JSON issue

### ISSUE THREAD

**Issue by vkulpa** [kensiprell/browser-websocket-client#6]
In our client/server WS communication we use partially valid JSON (e.g. `42["action",["event1","event2"]]`) I'd like to have ability to save such messages and preview pop-up with highlighted JSON.

I disabled options that validate JSON in messages (send/save) 
![disabled-options](https://raw.githubusercontent.com/vkulpa/browser-websocket-client/partially-valid-json-screenshots/screenshots/partially-valid-json/disabled-options.png)

but I'm not able to save it because it fails when I press "Save" button.
![try-to-save](https://raw.githubusercontent.com/vkulpa/browser-websocket-client/partially-valid-json-screenshots/screenshots/partially-valid-json/try-to-save.png)

Also, I'd like to have "preview" pop-up with JSON highlighted
![request-response-example1](https://raw.githubusercontent.com/vkulpa/browser-websocket-client/partially-valid-json-screenshots/screenshots/partially-valid-json/request-response-example1.png)
![request-response-example2](https://raw.githubusercontent.com/vkulpa/browser-websocket-client/partially-valid-json-screenshots/screenshots/partially-valid-json/request-response-example2.png)

**Comment by kensiprell** [comment]
If I push a new branch that addresses this issue, could you clone, build, and try it out in Chrome? The instructions to do that are in the README. Since this isn’t a feature I’d use, I’d like to get your feedback before I publish it.

**Comment by vkulpa** [comment]
Sure, thank you for the awesome plugin :-)

By the way, about PING/PONG - we could add section under options/preferences saying "enable PING/PONG messages", then in options/messages (or new dynamic section options/heartbeat) have text fields where PING/PONG message could be specified, as well as pingInterval etc.

**Comment by kensiprell** [comment]
Please clone branch 1.3.0, build it using `npm run build:chrome`, install it locally, and let me know if it meets your needs. I still have to fix failing tests and add new ones, so don't worry about those yet. I just want to make sure this is what you're after before going any further. It looks pretty simple, but I'm not sure how to improve it.

I'm still thinking about the ping/pong issue, but I think your idea above is the only way to do it.

Thanks for your help!

**Comment by vkulpa** [comment]
Thanks for the update. Save messages works fine now. :+1: 

Pop-up works differently than I expected. Might be misunderstanding. This is how it works now
![pop-up-with-invalid-json1](https://raw.githubusercontent.com/vkulpa/browser-websocket-client/partially-valid-json-screenshots/screenshots/partially-valid-json/pop-up-with-invalid-json1.png)
![pop-up-with-invalid-json2](https://raw.githubusercontent.com/vkulpa/browser-websocket-client/partially-valid-json-screenshots/screenshots/partially-valid-json/pop-up-with-invalid-json2.png)

The behavior I'd expect
![pop-up-with-invalid-json-desired1](https://raw.githubusercontent.com/vkulpa/browser-websocket-client/partially-valid-json-screenshots/screenshots/partially-valid-json/pop-up-with-invalid-json-desired1.png)
![pop-up-with-invalid-json-desired2](https://raw.githubusercontent.com/vkulpa/browser-websocket-client/partially-valid-json-screenshots/screenshots/partially-valid-json/pop-up-with-invalid-json-desired2.png)

In other words, I'd like to "try" the most outer valid JSON in response. `42["valid","json"]43` should be displayed and highlighted in pop-up as `["valid","json"]`. Every JSON starts with `[` or `{` and finishes with `]` or `}` so maybe recursively we try to find valid JSON starting from the most outer possible combination. Sound like a crazy requirement but I'd be happy if we just could skip some invalid part in the beginning. Hope examples help you to better understand my expectation.

Thank you in advance :-)

**Comment by kensiprell** [comment]
Thanks for the feedback, especially the images. I understand now what you’re looking for, but I want to spend a day or two thinking about it. Maybe there are other use cases I haven’t considered. I’ll probably ask you to take another look before I publish a new version, so stay tuned.

**Comment by kensiprell** [comment]
Please retry the latest 1.3.0 commit and let me know what you think. Most of the changes are here: https://github.com/kensiprell/browser-websocket-client/blob/9638f0b1f7988d864be1a7e68604f68bb3ad4ffa/src/main/index.js#L1044-L1080.

Here are some preview images:

![image](https://user-images.githubusercontent.com/480301/56457271-613b1c80-6378-11e9-8527-1ffede9e604c.png)

![image](https://user-images.githubusercontent.com/480301/56457274-6b5d1b00-6378-11e9-8bf3-60a489f81fb4.png)

![image](https://user-images.githubusercontent.com/480301/56457276-73b55600-6378-11e9-8880-091adb52baef.png)

![image](https://user-images.githubusercontent.com/480301/56457278-7b74fa80-6378-11e9-8270-2d9cc1b5b1ae.png)

![image](https://user-images.githubusercontent.com/480301/56457281-8760bc80-6378-11e9-9a38-f30238c1deab.png)

**Comment by kensiprell** [comment]
I just pushed a new commit for 1.3.0. It now looks like: https://github.com/kensiprell/browser-websocket-client/blob/09d4135d03569bc7c94c48d24f98723030cd1c0c/src/main/index.js#L1043-L1075

**Comment by vkulpa** [comment]
I've just checked. It works great I love it, thank you very much :+1:

**Comment by kensiprell** [comment]
Fixed in version 1.3.0.
